### ISSUE TITLE
Dont remove song from playlist if song is missing / removed from library

### ISSUE THREAD

**Issue by antipiot** [navidrome/navidrome#1426]
From time to time, i update the quality some albums and while doing so, they may be named a bit differently from the old one and this leads to the song being removed from the playlists he was appearing in

### Describe the solution you'd like

Songs in playlists missing in library appear as "missing" with a red cross or something like this, but never removed.

### Describe alternative solutions that would also satisfy this problem
A list of the removed song in order to rebuild the playlist

### Additional context

I've discovered recently this behavior and have now discovered that many updated songs are not in my plalists anymore wich is problem because they contain a lot of songs and i cannot retreive the missing ones from memory.

Thanks for your work and time!

**Comment by deluan** [comment]
Maybe it would be faster to just set `isMissing` to `true` (or `available` to `false`), whether or not the song is part of any playlist. This way, it would not cause any impact in the scanner (AFAICT).  We would also need a "Clean DB" option in the UI. 

Yeah, maybe this is worth to give a shot.

**Comment by certuna** [comment]
Yeah I like this - wouldn't take much space in the DB, and it would also be trivial to filter for recently deleted files etc.

**Comment by jkumeboshi** [comment]
What about try to lookup a song that was just moved in a different folder or renamed?
This may be achived through any hash function.

The photoprism web app has some similar feature, if you just move or rename a file, the albums containing it are automatically updated by the scan process.

**Comment by GlassedSilver** [comment]
> I always suffered with this as well, but this is a behaviour I copied from my main inspiration: macOS iTunes. And it is actually hard to keep the invalid song showing up in a playlist as all information from a deleted song is removed from the DB at scan time, and the playlist only stores a reference to it....
> 
> I'll keep this open, but I'm counting on the community to come up with some solution for this....

Well, if iTunes for Mac is your inspiration then you're probably also aware of the ingress concept of iTunes.

It scans for media, adds it to db along with paths and filenames and if they go missing it lets you relocate them.

Additionally, there's automatic organizing of media, so iTunes will place items in a file structure defined as `media type/artist/album/filename.ext`

That way manual look-ups usually are pretty quick, because at worst you recently moved your library to a new drive or something and some file didn't get moved for whatever reason and you go and manually fix it. (or it got corrupted and you restore from backup)

**Comment by deluan** [comment]
This is complete and released in 0.55.0. Check PID.Tracks configuration for customization: https://www.navidrome.org/docs/usage/pids/
